Detectors/TPC/reconstruction/src/TPCCATracking.cxx

```diff
@@ -248,28 +248,46 @@ int TPCCATracking::runTracking(const ClusterNativeAccessFullTPC& clusters, std::
         float zoffset = tracks[i].CSide() ? -tracks[i].GetParam().GetZOffset() : tracks[i].GetParam().GetZOffset();
         time0 = sContinuousTFReferenceLength - zoffset * vzbinInv;
```

Comment:
This time0 is then set to the track and used in the matching to define the time bracket for the track as
```
    float time0 = trcOrig.getTime0() - mNTPCBinsFullDrift;
    trc.timeBins.set(time0 - trcOrig.getDeltaTBwd() - mTPCTimeEdgeTSafeMargin,
                     time0 + trcOrig.getDeltaTFwd() + mTPCTimeEdgeTSafeMargin);
```
For the CCE tracks time0 calculated with request of the track converging to the vertex will be certainly wrong.
Should not the time0 in this case be calculated as  sContinuousTFReferenceLength + average time of CE-closest clusters?

Reply:
In principle, the tracker should already shift the track to exactly this offset, so this should not be needed. I'd rather have the correct shift already in the tracking.
However, I'd like to test this in O2. How can I simulate events with tracks crossing the CE?

Reply:
I mean the CE-crossing tracks cannot converge to Z=0 in principle (except for the strictly eta=0 tracks), so their shifting to Z=0 will move the clusters close to the CE to wrong side.
You should get such tracks in any simulation with non-0 vertex spread and smooth eta distribution, i.e. `o2sim -n50 -g pythia8` should produce them.